src/utils/label.ts

```diff
@@ -87,20 +90,44 @@ function hasAncestorCustomElements(label: HTMLCalciteLabelElement, componentEl:
 export function connectLabel(component: LabelableComponent): void {
   const labelEl = findLabelForComponent(component.el);
 
-  if (!labelEl || onLabelClickMap.has(labelEl)) {
+  const addClickEventListenerToComponentLabel = () => {
+    component.labelEl = labelEl;
+    const boundOnLabelClick = onLabelClick.bind(component);
+    onLabelClickMap.set(component.labelEl, boundOnLabelClick);
+    component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  };
+
+  const boundOnLabelConnected = onLabelConnected.bind(component);
+  const boundOnLabelDisconnected = onLabelDisconnected.bind(component);
+
+  if (onLabelClickMap.has(labelEl)) {
     return;
   }
 
-  component.labelEl = labelEl;
-  const boundOnLabelClick = onLabelClick.bind(component);
-  onLabelClickMap.set(component.labelEl, boundOnLabelClick);
-  component.labelEl.addEventListener(labelClickEvent, boundOnLabelClick);
+  if (labelEl) {
+    addClickEventListenerToComponentLabel();
+    unlabeledComponents.delete(component);
+    document.removeEventListener(labelConnectedEvent, boundOnLabelConnected);
+    document.body.addEventListener(labelDisconnectedEvent, boundOnLabelDisconnected);
+  }
+
+  if (!labelEl) {
+    unlabeledComponents.add(component);
```

Comment:
Just some cleanup...

`function onLabelDisconnected(this: LabelableComponent): void {` uses the same code as below. Maybe it could just be called here?